SRS/SRS_OC_BODBAW/srs_oc_bodbaw_netcdf_creation.py

```diff
@@ -0,0 +1,856 @@
+    var_depth        = output_netcdf_obj.createVariable("DEPTH", "f", "obs", fill_value=get_imos_parameter_info('DEPTH', '_FillValue'))
+    var_wavelength   = output_netcdf_obj.createVariable("wavelength", "f", "wavelength")
+
+    # wavelength
+    var = 'Wavelength'
+    wavelength_val_sorted = np.sort(np.unique(data.Wavelength))
+    var_wavelength[:]     = wavelength_val_sorted
+    if metadata['varatts_row'][var]['IMOS long_name']  != '':  setattr(var_wavelength, 'long_name', metadata['varatts_row'][var]['IMOS long_name'])
+    if metadata['varatts_row'][var]['Units'] != '':            setattr(var_wavelength, 'units', metadata['varatts_row'][var]['Units'])
+    if metadata['varatts_row'][var]['Comments'] != '':         setattr(var_wavelength, 'comments', metadata['varatts_row'][var]['Comments'])
+    if metadata['varatts_row'][var]['CF standard_name'] != '': setattr(var_wavelength, 'standard_name', metadata['varatts_row'][var]['CF standard_name'])
+
+    for var in np.unique(data.main_var_name):
+        output_netcdf_obj.createVariable(var, "d", ("obs", "wavelength"), fill_value=metadata['varatts_col'][var]['Fill value'])
+        if metadata['varatts_col'][var]['IMOS long_name'] != '': setattr(output_netcdf_obj[var], 'long_name', metadata['varatts_col'][var]['IMOS long_name'])
+        if metadata['varatts_col'][var]['Units'] != '': setattr(output_netcdf_obj[var], 'units', metadata['varatts_col'][var]['Units'])
+        if metadata['varatts_col'][var]['Comments'] != '': setattr(output_netcdf_obj[var], 'comments', metadata['varatts_col'][var]['Comments'])
+        if metadata['varatts_col'][var]['CF standard_name'] != '': setattr(output_netcdf_obj[var], 'standard_name', metadata['varatts_col'][var]['CF standard_name'])
+        setattr(output_netcdf_obj[var], 'coordinates', 'wavelength')
+
+    for i, var in enumerate(data.main_var_name):
+        var = data.main_var_name[i]
+        idx_wavelength = wavelength_val_sorted.tolist().index(data.Wavelength[i])
+        output_netcdf_obj[var][:, idx_wavelength] = np.array(data[i])
+
+    # Contigious ragged array representation of Stations netcdf 1.5
+    # add gatts and variable attributes as stored in config files
+    conf_file_generic = os.path.join(os.path.dirname(__file__), 'generate_nc_file_att')
+    generate_netcdf_att(output_netcdf_obj, conf_file_generic, conf_file_point_of_truth=True)
+
+    # lat lon depth
+    _, idx_station_uniq = np.unique(data.Station_Code, return_index=True)
+    idx_station_uniq.sort()
+    var_lat[:]          = np.array(data.Latitude)[idx_station_uniq]
+    var_lon[:]          = np.array(data.Longitude)[idx_station_uniq]
+    var_depth[:]        = data.Depth
+    var_depth.positive  = 'down'
+
+    # time
+    _, idx_time_station_uniq = np.unique(time_station_arr, return_index=True)
+    idx_time_station_uniq.sort()
+    time_values              = data.Dates[idx_time_station_uniq]
```

Comment:
Have you tried `to_pydatetime`? The pandas [docs](http://pandas.pydata.org/pandas-docs/stable/timeseries.html#converting-to-python-datetimes) seem to suggest that's what you want.